src/StreamJsonRpc/Reflection/RpcMarshalableKnownSubTypeAttribute.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace StreamJsonRpc;
+
+/// <summary>
+/// Attribute to be used on an interface marked with <see cref="RpcMarshalableAttribute"/> to indicate that a known
+/// inheriting interface exists.
+/// </summary>
+/// <remarks><para>
+/// When an object that implements <see cref="SubType"/> is marshaled as its base interface marked with
+/// <see cref="RpcMarshalableKnownSubTypeAttribute"/>, <see cref="SubTypeCode"/> is included in the StreamJsonRpc
+/// message allowing the creation of a proxy which also implements <see cref="SubType"/>.
+/// </para><para>
+/// If a message is received containing no <see cref="SubTypeCode"/>, for a marshalable interface that has known
+/// sub-types, a proxy will be created using the base interface. Unkonow <see cref="SubTypeCode"/> values will be
+/// ignored when creating the proxy.
+/// </para><para>
+/// <see cref="RpcMarshalableKnownSubTypeAttribute"/> is honored only when an object is marshaled through an RPC method
+/// that used exactly the interface the attribute is assigned to: <see cref="RpcMarshalableKnownSubTypeAttribute"/>
+/// attributes applied to interfaces extending or extended from the one the attribute is assigned to are ignored.
+/// </para></remarks>
+[AttributeUsage(AttributeTargets.Interface, AllowMultiple = true, Inherited = false)]
+public class RpcMarshalableKnownSubTypeAttribute : Attribute
```

Comment:
@AArnott, do we want to change the name of this attribute now that we allow adding interfaces don't extend the declared one?
Ideas:
- `RpcMarshalableKnownAdditionalInterfaceAttribute`
- `RpcMarshalableKnownInterfaceAttribute`
Or we can keep this name considering that most implementations will use subtypes of the declared interface.